InterfacePython/testOptimization.py

```diff
@@ -0,0 +1,112 @@
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+#    or promote products derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+# OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# UT-BATTELLE, LLC AND THE UNITED STATES GOVERNMENT MAKE NO REPRESENTATIONS AND DISCLAIM ALL WARRANTIES, BOTH EXPRESSED AND
+# IMPLIED. THERE ARE NO EXPRESS OR IMPLIED WARRANTIES OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE, OR THAT THE USE OF
+# THE SOFTWARE WILL NOT INFRINGE ANY PATENT, COPYRIGHT, TRADEMARK, OR OTHER PROPRIETARY RIGHTS, OR THAT THE SOFTWARE WILL
+# ACCOMPLISH THE INTENDED RESULTS OR THAT THE SOFTWARE OR ITS USE WILL NOT RESULT IN INJURY OR DAMAGE. THE USER ASSUMES
+# RESPONSIBILITY FOR ALL LIABILITIES, PENALTIES, FINES, CLAIMS, CAUSES OF ACTION, AND COSTS AND EXPENSES, CAUSED BY, RESULTING
+# FROM OR ARISING OUT OF, IN WHOLE OR IN PART THE USE, STORAGE OR DISPOSAL OF THE SOFTWARE.
+
+import unittest
+import Tasmanian
+Opt = Tasmanian.Optimization
+import os
+import numpy as np
+
+import testCommon
+
+ttc = testCommon.TestTasCommon()
+
+class TestTasClass(unittest.TestCase):
+    '''
+    Tests for the Tasmanian Optimization python bindings module
+    '''
+    def __init__(self):
+        unittest.TestCase.__init__(self, "testNothing")
+
+    def testNothing(self):
+        pass
+
+    def checkParticleSwarmState(self):
+        iNumDimensions = 2
+        iNumParticles = 5
+        # Initialization tests.
+        state = Opt.ParticleSwarmState(iNumDimensions, iNumParticles)
+        self.assertEqual(iNumDimensions, state.getNumDimensions())
+        self.assertEqual(iNumParticles, state.getNumParticles())
+        self.assertFalse(state.isPositionInitialized())
+        self.assertFalse(state.isVelocityInitialized())
+        self.assertFalse(state.isBestPositionInitialized())
+        self.assertFalse(state.isCacheInitialized())
+        # Loading/Unloading tests.
+        pp = np.random.rand(iNumParticles, iNumDimensions)
+        state.setParticlePositions(pp)
+        self.assertTrue(state.isPositionInitialized())
+        self.assertTrue(np.array_equal(pp, state.getParticlePositions()))
```

Comment:
There is a way to do this with one command, like assert almost equal, look for it among the other tests. It also works with given tolerance (could be zero).

Reply:
Changed to use `numpy.testing.assert_array_equal()`.